### PR TITLE
Added pricing floor support

### PR DESCRIPTION
**Pull request by pm-nikhil-vaidya**

### CODE REVIEW

File: config/config_test.go [hunk 1]
```diff
@@ -447,6 +447,10 @@ func TestFullConfig(t *testing.T) {
 	cmpInts(t, "gdpr.host_vendor_id", cfg.GDPR.HostVendorID, 15)
 	cmpStrings(t, "gdpr.default_value", cfg.GDPR.DefaultValue, "1")
 
+	//Asset the price floor values
+	cmpBools(t, "price_floors.enabled", cfg.PriceFloors.Enabled, false)
```
Comment (pm-jaydeep-mohite) [inline config/config_test.go:451]: Can we add test for checking cfg.PriceFloors.Enabled = true as well ?
Can be added as part of TestFullConfig

File: openrtb_ext/request.go [hunk 2]
```diff
@@ -56,13 +56,31 @@ type ExtRequestPrebid struct {
...
 	Transparency        *TransparencyExt    `json:"transparency,omitempty"`
+	Floors              *FloorExtension     `json:"floors,omitempty"`
 }
 
 type TransparencyRule struct {
 	Include bool     `json:"include,omitempty"`
 	Keys    []string `json:"keys,omitempty"`
 }
 
+type Schema struct {
```
Comment (pm-nilesh-chate) [inline openrtb_ext/request.go:67]: move this to `request_pubmatic.go` if the is OW fork exclusive code

**Comment by pm-nilesh-chate** [comment]
checklist from description is missing

File: config/config.go [hunk 3]
```diff
@@ -1101,6 +1107,8 @@ func SetupViper(v *viper.Viper, filename string) {
 	v.SetDefault("gdpr.tcf2.purpose_one_treatment.access_allowed", true)
 	v.SetDefault("gdpr.tcf2.special_feature1.enforce", true)
 	v.SetDefault("gdpr.tcf2.special_feature1.vendor_exceptions", []openrtb_ext.BidderName{})
+	v.SetDefault("price_floors.enabled", false)
+	v.SetDefault("price_floors.use_dynamic_data", false)
```
Comment (pm-nilesh-chate) [inline config/config.go:1111]: Do we need to explicitly set default value to false?

Comment (pm-jaydeep-mohite) [reply config/config.go:1111]: Yes

File: openrtb_ext/request.go (hunk 2 above)
Comment (pm-jaydeep-mohite) [reply openrtb_ext/request.go:67]: This in not OW specific, its part of Floors feature across prebid server applicable for all bidders.

File: config/config_test.go (hunk 1 above)
Comment (pm-nikhil-vaidya) [reply config/config_test.go:451]: Done

**Pull request closed without merging**